addons/spreadsheet_account/models/account.py

```diff
@@ -130,3 +158,28 @@ def get_account_group(self, account_types):
         )
         mapped = dict(data)
         return [mapped.get(account_type, []) for account_type in account_types]
+
+    @api.model
+    def get_residual_amount(self, args_list):
+        results = []
+        for args in args_list:
+            company_id = args['company_id'] or self.env.company.id
+            domain = self._build_move_lines_domain(args, error_domain=False)
+            [(amount_residual_currency_sum,)] = self.env["account.move.line"].with_company(company_id)._read_group(domain, aggregates=['amount_residual_currency:sum'])
+            results.append(amount_residual_currency_sum)
+        return results
+
+    @api.model
+    def get_partner_balance(self, args_list):
+        results = []
+        for args in args_list:
+            if args['partner_ids']:
+                company_id = args['company_id'] or self.env.company.id
+
+                domain = self._build_move_lines_domain(args, error_domain=False)
```

Comment:
Same, why don't we provide the partners and do a `groupby=['partner_id']` with `("account_id.account_type", "in", ("liability_payable", "asset_receivable"))`

If we want to be able to filter on accounts, why do we add a new function instead of just adding a parameter in `ODOO.BALANCE`